### PR TITLE
add slcan-support

### LINKED ISSUE #119: Windows compatibility for Lawicel CANUSB
Hello,

Is `python-can` compatible with [Lawicel CANUSB](http://www.can232.com/)? What configuration should I use (interface, channel)?

I'm running Windows python 3.5 x64.

Many thanks!

### PR DESCRIPTION
**Pull request by ebroecker**
add slcan/lawicel serial interface
these where supported with socketcan (slcand) already.
But this simple interfaces makes any slcan-compatible device usable for windows also

Note: no need to rename on of current modules any more.

Currently only tested with python2.7.

this should fix #119 and #5

please have a look on it.
I'll do further work on it, if needed.

### CODE REVIEW

File: can/interfaces/slcan.py [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+        super(slcanBus, self).__init__(channel, **kwargs)
+
+    def recv(self, timeout=None):
+        if timeout is not None:
+            self.serialPortOrig.timeout = timeout
+
+        canId = None
+        remote = False
+        frame = []
+        str = self.serialPort.readline()
```
Comment (christiansandberg) [inline can/interfaces/slcan.py]: I would suggest to not use `str` as it is defined by Python.

File: can/interfaces/slcan.py [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+            timeout in seconds when reading message
+        """
+
+
+        if channel == '':
+            raise TypeError("Must specify a serial port.")
+        if '@' in channel:
+            (channel, ttyBaudrate) = channel.split('@')
+
+        self.poll_interval = poll_interval
```
Comment (christiansandberg) [inline can/interfaces/slcan.py]: This is not used.

File: can/interfaces/slcan.py [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+                               extended_id=extended,
+                               timestamp=time.time(),   # Better than nothing...
+                               is_remote_frame=remote,
+                               dlc=dlc,
+                               data=frame)
+            else:
+                return None
+
+    def send(self, msg, timeout=None):
+        if bool(msg.is_remote_frame):
```
Comment (christiansandberg) [inline can/interfaces/slcan.py]: No need to convert to bool. Same for the other below.

File: doc/conf.py [hunk 4]
```diff
@@ -204,6 +204,6 @@ def __getattr__(cls, name):
         else:
             return Mock()
 
-MOCK_MODULES = ['serial']
+MOCK_MODULES = ['basic_serial']
```
Comment (christiansandberg) [inline doc/conf.py]: Is this change needed now?

File: can/interfaces/slcan.py (hunk 2 above)
Comment (ebroecker) [reply can/interfaces/slcan.py]: yes, you are right. 
poll_intervall is a copy-paste artefact
I'll remove

**Comment by ebroecker** [comment]
@christiansandberg 
All your commends should be fixed.

**Comment by christiansandberg** [comment]
Thanks, but the documentation must also be updated with the new interface.

**Comment by ebroecker** [comment]
some base documentation added

**Comment by ebroecker** [comment]
@christiansandberg 
Do I have anything else to do?

**Comment by christiansandberg** [comment]
Sorry. Update also doc/configuration.rst (and doc/history.rst if you want). Then I think you got the of it covered. :)

**Comment by hardbyte** [comment]
Thanks Eduard!
